### ISSUE TITLE
Quill editor settings being overridden by defaults due _.merge

### ISSUE THREAD

**Issue by rohanrichards** [formio/formio.js#4795]
https://github.com/formio/formio.js/blob/636685336c313bd8330d4e8c21a6720383b4b912/src/components/_classes/component/Component.js#L2209

As seen at the above line, a call to merge is used to place user settings over the top of the defaults, but in the case of Quill these settings are nested arrays and merging the two results in duplications/replacements where they are not wanted.

Example, a custom wysiwyg component with quill as the editor :
```
{
	weight: 0,
	type: 'textarea',
	editor: 'quill',
	wysiwyg: {
		modules: {
			toolbar: [
				[{ 'size': ['small', false, 'large', 'huge'] }]
			]
		}
	},
	label: 'Content',
	hideLabel: true,
	input: true,
	key: 'html',
	as: 'html',
	rows: 3,
	tooltip: 'The HTML template for the result data items.'
```

The above will result in the custom dropdown "size" being visible, as well as all of the defaults rather than the expected setup of the single specified toolbar item being visible. As seen in the below screenshot:

![Screenshot from 2022-07-28 16-10-32](https://user-images.githubusercontent.com/16222002/181433054-3862b5da-9a5a-4c08-b474-1595c9809251.png)

**Comment by daneformio** [comment]
Closing this thread as it is outdated. Please re-open if it is still relevant. Thank you for your contribution!

**Comment by Abdulllah** [comment]
@daneformio It's still relevant. I'd really appreciate it if you could re-open the issue and take another look.

**Comment by rohanrichards** [comment]
I'm not longer working on the project where this was relevant so I can not give any more information, sorry.

**Comment by Abdulllah** [comment]
@lane-formio I'm on version `4.14.9`, but the merge logic is the same in version `4.21.0`, so I think you can easily reproduce it there as well.

A side note: The result in this [example](https://formio.github.io/formio.js/app/examples/wysiwyg.html) is not what we would expect from the provided cusomt configuration:

```
      wysiwyg: {
        theme: 'snow',
        modules: {
          toolbar: ['bold', 'italic', 'underline', 'strike']
        }
      }
```

**Comment by lane-formio** [comment]
I've created a [sandbox here](https://formio.github.io/formio.js/app/sandbox.html).
Can you try loading the following id: 669ffe99c9b5abd4704c6163

Is this the expected behavior?

**Comment by Abdulllah** [comment]
@lane-formio Yes, the merge behavior in the sandbox is what we would expect from this configuration:

      wysiwyg: {
        theme: 'snow',
        modules: {
          toolbar: ['bold', 'italic', 'underline', 'strike']
        }
      }

**Comment by lane-formio** [comment]
It seems something is wrong/inaccurate on the example page. I'll get something on the backlog to investigate the example page but it looks to behaving as expected in practice so it will likely be very low in priority.

Please let me know if it seems I've misunderstood anything.

**Comment by jeriah-formio** [comment]
FIO-8738
